src/main/java/spoon/support/StandardEnvironment.java

```diff
@@ -353,13 +353,22 @@ public void setInputClassLoader(ClassLoader aClassLoader) {
 		if (aClassLoader instanceof URLClassLoader) {
 			final URL[] urls = ((URLClassLoader) aClassLoader).getURLs();
 			if (urls != null && urls.length > 0) {
-				List<String> classpath = new ArrayList<>();
+				// Check that the URLs are only file URLs
+				boolean onlyFileURLs = true;
 				for (URL url : urls) {
-					classpath.add(url.toString());
+					if( ! url.getProtocol().equals("file")) {
+						onlyFileURLs = false;
+					}
 				}
-				setSourceClasspath(classpath.toArray(new String[0]));
+				if( onlyFileURLs ) {
```

Comment:
As far as I understand the behavior that was previously implemented in setInputClassLoader, if the given class loader was an URLClassLoader, the URLs were extracted and transmitted as path elements to setSourceClasspath that is verifying with verifySourceClasspath that they are all correct path elements. If the given class loader is not an URLClassLoader, setInputClassLoader is not trying to extract path elements and simply sets the class loader in this.classloader.

The check you refer to in point 1 intents to implement a behavior that is as close as possible to the previous one. If the URLClassLoader contains anything that is not an URL that refers to a file, we revert to the second alternative (simply setting the class loader in this.classloader).

I do not mind if we choose to download URLs that are not file URL and extract their path from the locally downloaded version (your point number 2). This is a choice that needs to be agreed upon.